src/controllers/ElementsController.php

```diff
@@ -1054,7 +1058,21 @@ private function _errorSummary(ElementInterface $element): string
                             foreach ($tab->getElements() as $layoutElement) {
                                 if ($layoutElement instanceof BaseField && $layoutElement->attribute() === $fieldKey) {
                                     $tabUid = $tab->uid;
-                                    continue 2;
+                                    if (!$multiNested) {
+                                        continue 2;
+                                    }
+                                }
+                                // if it's a multi-nested error key for matrix in blocks mode
+                                // manipulate the key to only reference the matrix field, entry and inner field
+                                if ($multiNested && $layoutElement instanceof CustomField) {
+                                    if (
+                                        $layoutElement->getField() instanceof Matrix &&
+                                        $layoutElement->getField()->viewMode === Matrix::VIEW_MODE_BLOCKS
```

Comment:
@i-just Do you think it would be save to skip this check, so the same thing works for Neo? If it’s a Matrix field in cards/index view, then there won’t be any inputs to match for the error anyway.

Reply:
I tested that, and it won’t affect Neo.

Neo still uses [block indexes](https://github.com/spicywebau/craft-neo/blob/5.2.9/src/Field.php#L856) when prefixing model errors (switching to `$block->uid` makes the prefix look as expected). 
They’re also not using `errorKeyPrefix` when [rendering the blocks form](https://github.com/spicywebau/craft-neo/blob/5.2.9/src/templates/block.twig#L199), which means the error key is set to just the handle of the individual field (e.g. `myPlainTextField`). Passing `{'errorKeyPrefix' : "#{handle}[#{block.uid}]"}` as the 3rd param prefixes the errors as expected. 

With those two changes in place, anchoring to erroring fields inside Neo works as expected, and the lines you mentioned don’t affect that.